test/metabase/native_query_analyzer/replacement_test.clj

```diff
@@ -44,10 +48,20 @@
 
 (deftest ^:parallel field-filter-test
   (testing "with variables *and* field filters"
-    (is (= (str "\n\nSELECT *\nFROM folk\nWHERE\n  referral = {{source}}\n   OR \n  id = {{id}} \n  OR \n"
+    (is (= (str "SELECT *\nFROM folk\nWHERE\n  referral = {{source}}\n   OR \n  id = {{id}} \n  OR \n"
                 "  {{birthday}}\n  OR\n  {{zipcode}}\n  OR\n  {{city}} ")
-           (replace-names (q "\n\nSELECT *\nFROM people\nWHERE\n  source = {{source}}\n   OR \n  id = {{id}} \n  OR \n"
+           (replace-names (q "SELECT *\nFROM people\nWHERE\n  source = {{source}}\n   OR \n  id = {{id}} \n  OR \n"
                              "  {{birthday}}\n  OR\n  {{zipcode}}\n  OR\n  {{city}} ")
                           {:columns {"source" "referral"
                                      "city"   "town"}  ; make sure FFs aren't replaced
                            :tables  {"people" "folk"}})))))
+
+(deftest ^:parallel referenced-card-test
+  (testing "With a reference to a card"
+    (t2.with-temp/with-temp
+      [:model/Card {card-id :id} {:type          :model
+                                  :dataset_query (mt/native-query {:query "SELECT TOTAL, TAX FROM ORDERS"})}]
+      (is (= (format "SELECT SUBTOTAL FROM {{#%s}} LIMIT 3" card-id)
+             (replace-names (q (format "SELECT TOTAL FROM {{#%s}} LIMIT 3" card-id))
+                            {:columns {"TOTAL" "SUBTOTAL"}
+                             :TABLES  {"ORDERS" "PURCHASES"}}))))))
```

Comment:
Newbie question, why is `:TABLES` capitalized?

Reply:
`capitalize-region` gone wrong; fixed :facepalm: